pyproject.toml

```diff
@@ -31,4 +31,5 @@ build-backend = "setuptools.build_meta"
 optional-tests = [
   "no_python2: run when `python2` extra NOT installed",
   "no_blackd: run when `d` extra NOT installed",
+  "no_jupyter: run when `jupyter` extra NOT installed",
```

Comment:
So a thing with us maintainers is that we like to run bare `pytest $args` without using tox (mostly for speed and flexibility) and currently the test suite runs but then all of the Jupyter related tests crash with an ImportError as expected. It would be better if either a) test collection would fail immediately if the Jupyter dependencies aren't available, or alternatively b) juptyer tests could be automatically skipped (but that's unideal since it would be better to test as much as possible within a single `pytest` invocation).

For A, see what happens when the `d` extras aren't installed.

Reply:
That's a good point, thanks. I've addressed this so that `tests/test_ipynb` will not run if the extra dependencies aren't installed